### ISSUE TITLE
Tracking Issue for 0.7

### ISSUE THREAD

**Issue by diegogangl** [getting-things-gnome/gtg#737]
We're getting to the end of the 0.6 cycle, so it's a good time to start planning for 0.7 since we have lots of large, sweeping changes to land. This issue is meant to centralize, keep track of all the work and plan ahead so we don't end up with another dev hell scenario.

## Landing current PRs

- [ ] Land the new core PR #668 
- [ ] Land the Gtk4 port #723 
- [ ] Pin an issue mentioning the drag and drop regression, so we don't get extra tickets about that

## Polishing and completing the port

The new core was designed with Gtk3 in mind, with the idea to port over to listboxes and then onto Gtk4. @ranchester2 port changed the schedule a bit, so we have to move directly to Gtk4's listviews and models. We also need to finish some missing features.

- [ ] Add code for recurring tasks (@zeddo123 's territory)
- [ ] Check backends code with @jaesivsm 
- [ ] Port some properties in data classes to GObject properties (things that will need to be bound in widgets)
- [ ] Add a function to generate child models in store classes (using the children list)
- [ ] Add a model property to store classes
- [ ] Move the filtering  code to several `Gtk.Filter` subclasses in a their own modules (probably in a models folder)
- [ ] Add a `Gtk.Filter` subclass for search
- [ ] Move the sorting  code to several `Gtk.Sorter` subclasses in a their own modules too

## Plugging in the new core

Slowly slide in the new core so we can validate how it works

- [ ] Initialize `Datastore2` to `Application` and read the `gtg_data.xml` file
- [ ] Check the tasks are read correctly (using the dev console)
- [ ] Add a call to save with the new core in `requester.save_datastore`, set the path to `gtg_data2.xml`
- [ ] Check the xml files are the same
- [ ] Add calls to all the other parts of the requester , so both cores do the same (remove task, new task, etc.)
- [ ] Use the new core to setup the task editor, tag editor and batch tag editor

## Porting to listviews

There's a lot of new stuff regarding lists in Gtk4. I've written a quick 'n dirty (mostly dirty) script to test them out:
https://gist.github.com/diegogangl/eba75d467854068ea75cc9a03ce61495

This shouldn't include a lot of UI changes unless they are really simple, or necessary (like the new sorting)

- [ ] Design a UI for the new sorting, since we won't have treeview columns (kind of like the nautilus menu)
- [ ] Subclass Gtk.Box for each data type, so it can hold a pointer to the original object (_not sure this is the best way though_)
- [ ] Write the List item factory, DnD functions, etc
- [ ] Replace the treeviews in main window with a single listview, or 3 listviews (can we get away with just 1 with the pane system?)
- [ ] Figure out how to do the sidebar. Probably a `Gtk.ListBox` with rows for all tags/tasks with no tags, and 2 listviews for tags and saved searches mixed in the rows.
- [ ] Use the new filter based search
- [ ] Remove all calls and references to the old core
- [ ] Clean up whatever other mess we left behind

**Comment by diegogangl** [comment]
Made some mockups to sort out some ideas. I don't really know how Gtk4/libadwaita apps should look, but at least we can have a starting point:

![GTG_ui](https://user-images.githubusercontent.com/264829/147890661-ffb76bf1-9f5d-483b-83d4-105380e3325b.png)

SVG file if anyone wants to play with it:
https://mega.nz/file/Tx0iwCCZ#6qNAUNJuQwLQzxgYRif4h-kelLAtF_KyehbP8r1jvko

**Comment by zeddo123** [comment]
I'm not an expert in UI/UX, but I love it!
One thing though, apparently Glade won't provide Gtk4 widgets. [here](https://discourse.gnome.org/t/plan-about-gtk4-support-of-glade/5965)

**Comment by diegogangl** [comment]
Yup, we'll have to use [Cambalache](https://gitlab.gnome.org/jpu/cambalache) or write the XMLs by hand

**Comment by nekohayo** [comment]
Just to answer a question present in the mockup currently (since I'm not sure there's a ticket about that design specifically):
* I don't think it would look right if line heights vary, that's one of the things that annoys me quite a bit with Liferea's current "wide screen" layout mode, it just looks & feels broken/wrong.
* The ability to display 2 lines worth of content is interesting/nice-looking for those who don't have too many tasks, but if you have 500+ tasks, I don't think this will scale well especially since computer monitors are more height-constrained than width-constrained. It could make sense to display in two-lines mode when the current view (ex: selected tag) has few items in it, or when the screen is in portrait layout (as you'd get on a phone, or a portrait computer monitor) and auto-adjust to single line / compact mode otherwise when on a regular computer screen with lots of tasks...

**Comment by diegogangl** [comment]
Here's another take at the mockup, this time with single line rows. Maybe they could switch to 2 lines if the width is too small (though I don't know if this actually possible). 

Let's see if I picked the right icons for start and due :)

![single_row](https://user-images.githubusercontent.com/264829/159981792-78a3b9a5-a7a0-4d51-b51c-4d3b052a4310.png)

**Comment by nekohayo** [comment]
Also, since you mentioned libadwaita, and your mockups kind of resemble its flat styling (but I only considered them to be wireframes more than high-fidelity mockups): if at all possible, please let's try to stick to pure GTK4 (and see how far we can go with that), and hold back from using libadwaita for the foreseeable future, if we can. I personally think libadwaita's "trendy" flat aesthetic is a regression in usability (I want my buttons to look like damned _buttons_ that can be interacted with, not every widget component looking like a low-contrast solid-filled gray rounded rectangle!)

While I suspect I might be part of a minority, I surely am not alone to think that way (some people did [complain about its buttons' usability issues](https://gitlab.gnome.org/GNOME/libadwaita/-/issues/228), me included), and unless some drastic changes occur in the design team's orientation (which sounds really unlikely to me), I wouldn't really want to enforce that flat style onto our users unless we have a good technical reason to do so.

**Comment by environmentalcode0** [comment]
I couldn't disagree more with @nekohayo. I don't think this "trendy" aesthetic is a regression -- I personally love it -- even though this may be something subjective and I understand your reasons. But as a user I try to only use GTK4+libadwaita apps (I don't even know if there GTK4 ones without libadwaita) for a more consistent look and I wouldn't want GTG to remain so old looking.

**Comment by nekohayo** [comment]
Well, disregard my personal "substance over style" IxD opinions on widgets usability for a minute, and consider my above argument from a multi-year strategic project management worldview: _if_ we can achieve most/all that we want with pure GTK4, I would like to avoid rushing into adopting the extra dependency that is libadwaita; this may avoid painting ourselves into a corner technically, possibly save us some work/rework if anything happens to be short-lived in the stack we depend on (oh, how many times have I seen graphics-related libraries come and go!...), and possibly avoid drama around the whole styling thing; for users who would be unhappy with the app looking "not flat enough", I would think they would already set a flat libadwaita-like theme on their system that would apply to all their pure-GTK4 apps anyway, no?

That said, maybe fellow GTG contributors, upon technical evaluation, will overwhelmingly tell me that the functionality we need really requires libadwaita, at which point my argument would become moot, of course. I'm not sure we need all that to happen in the 0.7 cycle (especially if we want to get 0.7 out faster) unless some jaw-dropping UX-enhancing features in GTG are already made and strictly depend on that library (so far I can't imagine what they could be, however).

**Comment by diegogangl** [comment]
I just grabbed an SVG file I found on Gnome's gitlab :)
I'm a bit torn on libadwaita. It does enable adaptative UIs, automatic dark style and some other cool things. On the other hand, it apparently makes user theming harder (since it hardcodes the stylesheet), adds an extra dependency and I'm not entirely sold on the new stylesheet. 

However, we could add a background to buttons. Timetrack does this:

![Screenshot from 2022-03-26 19-40-01](https://user-images.githubusercontent.com/264829/160259316-ed5ced0c-04a7-4b52-82f9-a65fded5a793.png)

In any case, there are no plans for/against libadwaita yet. The important thing in the mockups is the task list design and the sidebar

**Comment by thegranddesign** [comment]
From what I gather libadwaita is only a layer on top of GTK in a way to make GTK apps consistent in look and feel.  I have to say that if the current look and feel of GNOME (eg files, calendar, etc) is in any way related to libadwaita then that should be the standard set for GTG.  Incredible kudos to everyone working on this project but the current look and feel of GTG looks as if a developer designed it, which totally makes sense due to the history, but doesn't give GTG a "polished" feel.

@nekohayo I totally get what you're saying about buttons, etc but to be honest the web has already taught everyone that not only buttons are clickable.  If I see that tag list with a count next to it in a sidebar, I know it's clickable regardless of what it looks like.  If I want to be certain, I hover my mouse over and see if it has a hover effect.

At the end of the day, consistency with the rest of the apps being built for GNOME is *significantly* more important than doing what feels "right".  If everyone did the thing that they "liked best" then we'd have apps that look completely disjointed.  There's a reason why Apple has their style guide and everyone adhered to it.  Everyone using their apps knew how their apps worked because they all looked the same.

I would also add that I think the architecture of the data should map to the UI first and foremost.  I don't think we should design the data architecture and then back into the UI.  I think we should design the UI and then determine the data architecture that makes those things possible (potentially future-proofing with some additional ideas).  For example, do we really need infinitely recursable/nested tasks?  Might it not be better to use a standard Project/Task architecture?  Is the ability to nest infinitely conducive to "getting things done"?

I would also look at OmniFocus.  Their UI is crap, but they do have a few concepts in their tasks and projects that are nice to haves (especially around recurrence).

@diegogangl I think your mockups look freaking amazing. :heart_eyes:   One thing I would caution you on is the propensity to think of the data in mockups as "sacred".  When designers do mockups they think in "perfect world" terms.  I would challenge you to do:

1. a mockup that has 50 tasks and some of those tasks have > 10 levels of nested tasks (even though I think deep nesting is a bad idea, it's possible today and so needs to be considered)
2. put 50 tags in the tag list (some with extremely long names)
3. What does the "saved search" look like if it has a very long name?
4. How would a task count of 9,999,999 be handled?

Doing these "real world" but "imperfect" things will make a mockup able to be judged in a better light.  I think the primary issue you'll find with the current mockup layout (similar to what @nekohayo pointed out) is that all of that padding is going to be a problem.  I agree that there needs to be _some_ breathing room between the items (versus current GTG's list view) but what you have there is (in my opinion) more than what will be desirable for actual use cases.

**Comment by thegranddesign** [comment]
Additionally I would point out that using libadwaita allows GTG to leverage changes made by the entire community rather than having to do all of that work itself.  If the libadwaita community eventually decides that buttons styled differently work better for a certain reason then GTG doesn't have to re-implement that work.  Again, I don't think anyone on this team is trained as a designer, so let the designers do what they do best and the developers can do what they do best.

**Comment by diegogangl** [comment]
Hey, sorry just saw your replies. Thanks for the thorough comments!

> At the end of the day, consistency with the rest of the apps being built for GNOME is significantly more important than doing what feels "right". If everyone did the thing that they "liked best" then we'd have apps that look completely disjointed. There's a reason why Apple has their style guide and everyone adhered to it. Everyone using their apps knew how their apps worked because they all looked the same. (And similarly with Google's "Material" style.)

Good point

> I would also add that I think the architecture of the data should map to the UI first and foremost. I don't think we should design the data architecture and then back into the UI. I think we should design the UI and then determine the data architecture that makes those things possible (potentially future-proofing with some additional ideas). For example, do we really need infinitely recursable/nested tasks? Might it not be better to use a standard Project/Task architecture? Is the ability to nest infinitely conducive to "getting things done"?

I'm sure @nekohayo can come up with examples :)
I think infinite nesting will work very nicely once we have a third pane to show the task contents in the main window. You can basically treat a task as a project. Subtasks can be project steps, or entire subprojects themselves. In other words, it would be flexible enough that you can scale from "walking the dog" to "build fission reactor" with the same UI.

> @diegogangl I think your mockups look freaking amazing. :heart_eyes:  One thing I would caution you on is the propensity to think of the data in mockups as "sacred". When designers do mockups they think in "perfect world" terms. I would challenge you to do:

Absolutely!  I'm also the one writing the code, so I'm aware of how rough a sketch this is :stuck_out_tongue: 

> a mockup that has 50 tasks and some of those tasks have > 10 levels of nested tasks (even though I think deep nesting is a bad idea, it's possible today and so needs to be considered)
> put 50 tags in the tag list (some with extremely long names)
> What does the "saved search" look like if it has a very long name?

If Gtk allows we should probably lower the indent for subtasks/tags. Super deep nesting is definitely something to consider in the future, because I'm not sure what solutions are possible (extending the rows to add a horizontal scrollbar, hiding certain elements, etc). Long names would be ellipsized (I think Gtk already does automatically).

> How would a task count of 9,999,999 be handled?

Like this -> :bomb: :boom: 

> Doing these "real world" but "imperfect" things will make a mockup able to be judged in a better light. I think the primary issue you'll find with the current mockup layout (similar to what @nekohayo pointed out) is that all of that padding is going to be a problem. I agree that there needs to be some breathing room between the items (versus current GTG's list view) but what you have there is (in my opinion) more than what will be desirable for actual use cases.

I don't remember if that's from the original mockup or I went overboard myself. The new sidebar code so far has less padding.

> Additionally I would point out that using libadwaita allows GTG to leverage changes made by the entire community rather than having to do all of that work itself. If the libadwaita community eventually decides that buttons styled differently work better for a certain reason then GTG doesn't have to re-implement that work. Again, I don't think anyone on this team is trained as a designer, so I think we should let the designers do what they do best and the developers can do what they do best.

I'm actually trained as a graphic designer and did web design for several years :sweat_smile: @nekohayo does branding, filming and witty writing. The main problem with GTG (whether UI or backend code) is that we are carrying a lot of legacy, and we're all swamped with work and other things, so things improve rather slowly and depend on massive rewrites (like the #894 PR) .

**Comment by nekohayo** [comment]
Just selectively replying to a few of @thegranddesign's statements there, because I don't have much time nor energy lately:

> I totally get what you're saying about buttons, etc but to be honest the web has already taught everyone that not only buttons are clickable. If I see that tag list with a count next to it in a sidebar, I know it's clickable regardless of what it looks like. If I want to be certain, I hover my mouse over and see if it has a hover effect.

_You_ know this, and geeks and "digital natives" know this, but my 50-80-years-old family members don't. Some business owners I see using computers, also, are absolutely excruciating to watch. They reveal all the flaws of the modern style-over-substance UIs. I have thousands of UI design issues filed across GNOME and the freedesktop to prove my point.

I suspect you haven't spent enough years doing actual in-the-trenches usability studies and tech support to realize just how disconnected from the average non-techie layperson you can be. I did, and still do such observations these days; that's why I find the current incarnation concerning visually.

But my point is also that, just technologically+technically speaking, if we don't strictly need to depend on a _specific feature_ of libadwaita beyond the standard GTK widgets, I don't see why we should go through that "just because", sounds to me like painting ourselves in a corner. It has to be a very clear win from a maintenance perspective vs the cost of potentially needing to rip things out some years down the road (if I had a dollar for everytime someone wanted to switch canvas toolkits in some of the desktop projects I've been involved in...)


> [...] I don't think anyone on this team is trained as a designer, so I think we should let the designers do what they do best and the developers can do what they do best.

Just to rectify the record here:

I actually _am_ an interaction designer (I even have [a hidden page about that](https://fortintam.com/en/interaction-design)), and usability-focused interaction design has been my primary way to contribute to FLOSS over the last ~20 years (along with QA and project management); I very seldomly develop code myself (I just know an abnormal amount of coding tricks to be able provide occasional guidance, or to know when something is/isn't trivial). In my day-to-day, my clientèle's needs are typically _much_ higher-level than UI design, so I don't emphasize my design work much at all, because the value I provide in business management consulting completely dwarfs any application-level design work I do. Just because I have philosophical disagreements with the "flat" _fast fashion_ doesn't make me clueless about modern design patterns ;) nor do I have final say in design matters here; at the end of the day, this is a community/patches-driven project, to which I just happen to provide opinions, guidance and support when time allows, hopefully preventing pitfalls projects commonly fall through.

**Comment by thegranddesign** [comment]
@nekohayo I'm just providing my 2 cents.  I don't know Python so I can't contribute and am not doing to judge the final decisions made here.  As someone who has a couple very popular libraries I know the burden of maintaining for free.

A couple points and I'll let this drop (unless @diegogangl has another amazing mockup to show :heart_eyes:)

> the "flat" _fast fashion_

I'm not exactly what you consider "fast".  Flat has been around in major operating systems for over 10 years.

> You know this, and geeks and "digital natives" know this, but my 50-80-years-old family members don't.

I don't see 50-80 year old non-techies adopting Gnome any time in the next couple decades

> vs the cost of potentially needing to rip things out some years down the road (if I had a dollar for everytime someone wanted to switch canvas/UI/style toolkits in some of the desktop projects I've been involved in...)

And if I had a dollar for every time someone recreated an already existing abstraction because it "wasn't invented here"... :wink: 

Regarding infinite recursion, I still don't think it's necessary.  I think trying to design a UI that is good for everything means it's best at nothing.  Tasks are not projects.

Again, just my two cents.  I'll be using GTG regardless because it's the best open source thing currently out there. :heart: 

Keep up the great work everyone!

**Comment by nekohayo** [comment]
@diegogangl I don't know if this changes anything to your plans (or if that is already where you were headed), but in case it is somehow insightful or inspiring, the Nautilus maintainer [blogged](https://blogs.gnome.org/antoniof/2022/06/15/the-tree-view-is-undead-long-live-the-column-view%e2%80%bd/) about his experience with migrating away from TreeView and to ColumnView, which apparently can also still do tree views (?!)...

**Comment by diegogangl** [comment]
@nekohayo ended up replying late, but yeah we can still do trees with `Gtk.TreeListModel` and `Gtk.TreeExpander`. We are already using those in the megaport. There's only one issue with hiding/showing the expander arrow dynamically. That's been solved (I think) in Gtk master

**Comment by diegogangl** [comment]
@gargullia Thanks for the kind words. I'm leaning torwards libadwaita, but definitely not for 0.7. We're _way_ over our heads with all the changes (you can check the #894  PR)

**Comment by gargullia** [comment]
@diegogangl You can also use [Blueprint](https://jwestman.pages.gitlab.gnome.org/blueprint-compiler/)

Information from here: https://thisweek.gnome.org/posts/2022/09/twig-61/

`Bottles was recently [ported](https://github.com/bottlesdevs/Bottles/commit/94bc8ffb73fa9bbd979e11cef10f884e716ef790) to [Blueprint](https://jwestman.pages.gitlab.gnome.org/blueprint-compiler/)! Blueprint is a new markup language to create GTK interfaces. It is more efficient for development as the syntax is less verbose and more visually pleasing in contrast to XML. Thanks to its efficiency, Bottles’s codebase was reduced by 800 lines!`

**Comment by nekohayo** [comment]
The big refactoring has finally landed! I have updated the description above, clarifying the next steps and providing some convenient links.

Anyone who may have been waiting to land their code: please rebase your branches to the latest `master`.  
The new codebase is expected to be much better to work with.

**Comment by pentux-GitHub** [comment]
Quick status of the review queue from my side, since several things
landed today that touch the 0.7 milestone:

- #1297 — CalDAV: push local deletions to the server (follow-up to the
  #1265 scope note). CI green, live-tested against Nextcloud.
- #1298 — heal stores whose tasks have no added date. CI green; merging
  it auto-closes the critical #1033 (prevention half already landed
  with #1265, details on the issue).
- #1294 — sync-now button: waiting on your icon call (`update-symbolic`
  doesn't exist under that name anywhere I could find — two real
  candidates proposed in the thread).
- #1240 — can be closed, superseded by #1265 (comment posted there).

Also worth noting for the milestone: #1283 and #1277 were already
reviewable before all this and would close #1274 and #1001
respectively.

No rush — just putting the map in one place.

**Comment by pentux-GitHub** [comment]
Thanks to everyone who carried 0.7 this far. We're almost there: what's mainly left is the performance work (#109, #840).

And if you use GTG like I do most days, with lots of tasks, your numbers on what feels slow would really help for the final sprint. Anyone up for it?

Let's finish this together. 🙂

**Comment by pentux-GitHub** [comment]
With the CalDAV backend now ported to the new core and running daily against a real Nextcloud instance, here is an overview of where it stands for 0.7, and a suggestion for closing the cycle cleanly. I contributed to this work, so I feel responsible for stating plainly what is solid and what is not.

What ships solid: the backend runs on the new stores (#1265), non-UUID identifiers no longer cause data loss (#1303), the category to tag projection is reversible again and hardened against special characters (#1306), the 0.6 Dismiss status survives migration (#1309), a Synchronize now button exists (#1294), and the recurrence groundwork is in place with round-trip tests (#1304). In daily use, tasks, hierarchy, tags and statuses sync reliably.

Three small items could still land before the release, if you agree. First, #1296: the internal DAV_ calendar tags are exposed as ordinary user tags, and since the backend routes writes through them, renaming or deleting one silently breaks the routing for that task. Hiding them by default, with an option to show them, is the same size of change as #1311 and I can have a patch ready quickly once the approach is validated. Second, the Deck calendars from Nextcloud produce harmless perpetual diffs due to their non-UUID stack identifiers; nothing to fix, but one line in the docs or release notes would spare users the worry. Third, the Flathub manifests may need a look regarding the pinned caldav version versus what distributions ship; I can provide the exact current state if useful.

Some known limitations are, in my view, better deferred than rushed. Local task deletions are not propagated to the server yet (disclosed in #1265); wiring automatic remote deletion close to a release is exactly the kind of change that deserves design and long testing rather than haste. Data files that synced before #1306 can contain mangled category names that keep producing diffs; fixing that means rewriting stored user data, same caution applies. And recurrence sync depends on the design discussion in #645.

So the suggestion is simply this: 0.7 ships CalDAV as ported, hardened on data integrity and usable daily, with these limitations named in the release notes, and a small follow-up cycle, whether that is a 0.7.1 or early 0.8 is entirely your call, closes the remaining three. I am happy to keep carrying this work either way.

**Comment by pentux-GitHub** [comment]
Two corrections and a suggestion, following an end-to-end pass over the milestone today.

**Correction to my wrap-up above**: I listed "local task deletions are not propagated to the server" as a known limitation to defer — that's outdated, and by my own doing: #1297 landed on July 21 (commit `f015fa04`, still in master) and does exactly that, and my wrap-up was posted on July 30. So the CalDAV backend for 0.7 ships *with* local deletion propagation, and the release notes should say so rather than the opposite. Sorry for the confusion.

**The Flathub/caldav state I promised**: the dev manifest pins `caldav==1.4.0`. While working on #1312 I inspected the `calendar.todos()` signature and sort guard on caldav 0.11.0, 1.4.0, 2.0.1 and 3.2.1, and verified the behavior empirically on 3.2.1: stable throughout, so the pin poses no compatibility problem on that axis. (I haven't audited the rest of the manifest.)

**A suggestion: give the release notes a home now.** Several "one line in the release notes" promises have piled up across threads with no place to land: the harmless perpetual diffs on Nextcloud Deck calendars (my wrap-up above), the sidebar showing all historical tags after a 0.6 migration (#1310, with #1311 as the opt-out), the CalDAV limitations that *do* remain (mangled pre-#1306 category names, recurrence pending #645), and the correction above. A draft release-notes issue or document would catch these before they evaporate — happy to write the first draft if that helps.

**Description refresh**: the checklist above predates the summer: the CalDAV backend (#1265 and follow-ups) and the Hamster plugin (#1287) are ported, and the post-#894 fallout review has been substantially done through the regression wave (#1252 being the notable regression still open). @nekohayo if you'd like, I can propose updated wording for those sections for you to paste in. With #737 closing at tag time and #939 at Flathub release time, the milestone effectively comes down to the two performance issues, #109 and #840.

**Comment by pentux-GitHub** [comment]
As promised, here is suggested wording to refresh the outdated sections of the description — ready to paste:

**Section "Revamping the plugin and backends API, and fixing plugins"**, replace the 2024 status block with:

> **2026 status:** the CalDAV backend was ported to the new core and hardened through a dedicated summer campaign (#1265 and ~15 follow-up PRs: local deletion propagation, reversible tag projection, non-UUID id support, Synchronize-now button, migrated test suite). The Hamster plugin was ported in #1287, and a failing plugin can no longer prevent startup (#1286). The backends-vs-plugins API question remains open and is 0.8 material.
> - [x] ~~The most important of CalDAV-related issues~~ — remaining CalDAV items are design RFCs (#1291, #1296, #1305) and features tracked for 0.8
> - [x] ~~The most important of Hamster-related issues~~ — #996, #1139, #993 are resolved pending closure

**Section "Overall Status"**, the "Review and fix the fallout" item can be checked:

> - [x] Review and fix the fallout — done through the 2026 regression wave (see the ~40 issues closed in the milestone since June); #1252 is the notable regression still open

And a pointer worth adding at the top of the description: release notes are now being drafted in #1318, so anything worth telling users at tag time has a place to land.

**Comment by pentux-GitHub** [comment]
A tracker housekeeping proposal, round one — the most factual batch. With 183 open issues, some cleanup would make the real work easier to see. Every item below is closable based on a verifiable fact, not an opinion (a one-line rationale is posted on each issue); of course the final call is yours, and anything can be reopened.

**Already labeled as closed-worthy by maintainers:**
- #1203, #1237 — both carry the `duplicate` label
- #523 — carries the `wontfix` label

**Reports against backends that are no longer shipped** (RTM and Mantis live in `backends/unmaintained/`, not loaded, not shipped):
- #58, #63, #64, #65, #75, #201 (Remember the Milk)
- #157 (Mantis)

**Reports about a dependency that no longer exists** (the new core removed liblarch entirely):
- #709, #321

That's 12 issues, 183 → 171. It seems safe to close all of them, and I can prepare round two (issues absorbed by the #645 recurrence RFC, verified-fixed items, and stale `needinfo` reports) if this one lands well.

**Comment by pentux-GitHub** [comment]
Tracker housekeeping, round two — this one backed by actual testing on current master tonight, not just labels.

**Verified in the code:** #1051 (datasets are copied), #292 (the parenting signals exist), #670 (column tree views no longer exist), #220 (GTK 3 specific)

**Explicitly absorbed by the recurrence RFC #645:** #627, #614, #636, #626 (each is a named bullet or section of the RFC)

**Tested tonight on master, resolved:** #993 (Hamster config opens), #906 (F1 help works), #1041 (UI fully translated in French), #900 (works as designed)

**Unactionable** (empty tracebacks, info never provided, code rewritten since): #1257, #1268, #1269, #713

Individual rationales are posted on each issue. That's 16 more, which together with round one would bring the tracker from 183 to 155. Also of note: while testing, #940 turned out to be **still alive** on master (repro posted there) — it graduates from stale candidate to fix candidate.

Round three (the last one: a handful of GTK3-era UI reports needing individual retests) can follow if this lands well.
